documentation/modules/exploit/multi/http/gitlab_exif_rce.md

```diff
@@ -0,0 +1,262 @@
+## Vulnerable Application
+
+### Description
+
+This module exploits an unauthenticated file upload and command injection vulnerability in GitLab Community Edition (CE) and
+Enterprise Edition (EE). The patched versions are 13.10.3, 13.9.6, and 13.8.8.
+
+Exploitation will result in command execution as the git user.
+
+### Installation
+
+#### GitLab 13.10.2 on Ubuntu 20.04.2 x64
+
+* Download [GitLab 13.10.2](https://packages.gitlab.com/gitlab/gitlab-ce/packages/ubuntu/focal/gitlab-ce_13.10.2-ce.0_amd64.deb)
+* Install openssh-server (`sudo apt install openssh-server`)
+* Install GitLab (`sudo dpkg -i gitlab-ce_13.10.2-ce.0_amd64.deb`)
+* Modify the `external_url` in `/etc/gitlab/gitlab.rb` to something like `external_url http://localhost`
+* Run `sudo gitlab-ctl reconfigure`
+* Done!
+
+## Verification Steps
+
+* Follow the above instructions to install GitLab 13.10.2
+* Do: `use exploit/multi/http/gitlab_exif_rce`
+* Do: `set RHOST <ip>`
+* Do: `check`
+* Verify the remote target is flagged as vulnerable
+* Do: `set LHOST <ip>`
+* Do: `set LPORT <port>`
+* Do: `exploit`
+* You should get a Meterpreter session.
+
+## Targets
+
+### 0
+
+By default, this targets GitLab with the `reverse_openssl` payload and returns a reverse shell.
+
+### 1
+
+By default, this target obtains a meterpreter session using `wget`. This target also supports
+`lwprequest`, `curl`, and `printf`. However, due to the exploit's space constraints it's not
+recommended to use a meterpreter payload with `printf` due to the time it will take to write the payload.
```

Comment:
It's probably taking longer because you're using a stageless payload.

Reply:
FWIW, it may be helpful to default to a staged payload for maximum `CmdStager` efficiency.